### ISSUE TITLE
Ability define test management interface for meshconfig agent defined measurements

### ISSUE THREAD

**Issue by igarny** [perfsonar/mesh-config#42]
Dear pS Dev,

Please consider the following use case:
In an isolated environment (LHCONE) it would be nice (important for GEANT) to be able to instruct meshconfig agent how to organize the coming measurements.

Use case is: 

- the default management interface with default GW (defGW) is connected to the global Internet connectivity.
-  the test interfaces are connected to an isolated VRF (LHCONE)

In this case we need to have the ability to instruct pscheduler how to coordinate the tests, since current implementation of pscheduler uses the default GW interface to start test negotiations.
There are also valid use cases, where it is valuable to separate test negotiation outside the measurement communication.

IMHO it would be quite a burden for the mesh administrator to be aware of the peculiarities of each server that is joined to a mesh (consider the WLCG use case).
IMHO the local admin is fully aware of the server connectivity specifics, thus she should be able to define these in local config files.
My proposal is:
- to implement a parameter in /etc/perfsonar/meshconfig-agent-tasks.conf, that will instruct pscheduler to use the local interface listed in the meshconfig for test negotiation openings.
- Also to have a parameter in the /etc/perfsonar/meshconfig-agent.conf to instruct meshconfig service to populate this parameter in the "....tasks.conf" file

regards,
Ivan

**Comment by igarny** [comment]
Dependent on perfsonar/pscheduler#234

**Comment by igarny** [comment]
Hi Andy,

As discussed by emails and/or VC. The implementation here would follow the -B solution for BWCTL cases with pS 3.5.
I hope it would not make it too complex to provide 3 options here (2 defaults) for the selection of the default interface:
1) default: (same as omitting the key). Having a default statement would ease automated parsing of config.
2) test_interface (or whatever keyword you come up with): to make meshconfig and pscheduler override the default routing and use/apply the address from the test specification. After all one of the addresses in the test specification, should be local. 
    I believe this would be the default for all multihomed systems, unless, for example, one wants to have the throughput tests negotiated over the latency interface (in the case with default GW on latency in-face) or vice-versa.
3) IP address / FQDN: to make meschonfig and pscheduler override the default routing and use the specified interface as binding interface on pscheduler tasks.

**Comment by arlake228** [comment]
Figured out there is another set of options for grabbing results, so also need to set --source-node and --dest-node on throughput tests. looks like this requires another set of options. From a parsing perspective works essentially the same as local_lead_node and lead_node, just it gets mapped into --source-node and --dest-node....also need to make sure I name it something not too confusing.

**Comment by arlake228** [comment]
Looks like a lead-bind GET parameter got added to the participants method which mesh-config calls when determining the lead. Need to modify it to pass this in for it to work properly.

**Comment by arlake228** [comment]
Added fix in perfsonar/perl-shared@ba9ad7fd24d74557e6e24535f82bd61b27f9b4b8 to set lead-bind in GET parameters to participants. Manually patched psps-test2-mgmt.rrze.uni-erlangen.de and rpm building now. initial results look promising. Primarily, there is now a task at: https://psps-test2-mgmt.rrze.uni-erlangen.de/pscheduler/tasks?json={%22reference%22:{%22created-by%22:{%22address%22:%22psps-test2-mgmt.rrze.uni-erlangen.de%22}},%22test%22:{%22type%22:%22throughput%22,%22spec%22:{%22dest%22:%22perfsonardev0.internet2.edu%22}}}

**Comment by arlake228** [comment]
Looks like lead-bind worked when I added it to the central mesh file. It did cause an issue for 3.5 hosts simply because it was unrecognized option, which is true of any new option and not specific to the bind address. On the 4.0 side though all the boxes to perfsonardev0 filled-in when I had it in there.

After that ran for awhile, I removed the bindings from the central mesh file so the 3.5 hosts could work again. I added bind_address and local_lead_bind_address to the top of the meshconfig-agent-tasks.conf file by hand and that seems to work for anything initiated by the rrze.uni.erlanged.de host. Since we run force_bidirectional there will still be blue boxes for the throughput tests initiated by perfsonardev0 since it does not know what the lead-bind should be, but that's expected. 

Closing this since I am comfortable it is working the way it should.
